ui/src/source.ts

```diff
@@ -140,24 +163,56 @@ export const commits = commitsStore.readable;
 const objectStore = remote.createStore<SourceObject>();
 export const object = objectStore.readable;
 
-const revisionsStore = remote.createStore<Revisions>();
+// Updated when the project store changes (i.e. when the user navigates to a new project)
+const revisionsStore = remote.createStore<PeerRevisions[]>();
 export const revisions = revisionsStore.readable;
+revisions.subscribe(store => {
+  if (store.status === remote.Status.Success) {
+    // The first `PeerRevisions` in the response belongs to the default peer.
+    const defaultRevisions = store.data[0];
+
+    const peerId = defaultRevisions.identity.peerId;
+    updateCurrentPeerId({ peerId });
+
+    // Now that we have a peer, set the current revision to the default branch.
+    // If not found, use the first branch returned from proxy.
+    const defaultRevision =
+      defaultRevisions.branches.find(
+        branch => branch.name === defaultRevisions.defaultBranch
+      ) || defaultRevisions.branches[0];
```

Comment:
And in the event that there are no branches?

Reply:
In this case, the revision selector won't show up but the project will still render. Before, the `RevisionSelector` would create a default revision that may or may not actually exist (https://github.com/radicle-dev/radicle-upstream/pull/995/files#diff-ac6e05ed01605cc8ea4ba88d402e6541L30).

Should we throw an error instead? I don't really like the idea of creating a revision in ui that we don't actually receive from proxy.

Reply:
Ya, I definitely agree that creating a made-up revision doesn't make sense. Could we have like an error window in the selector saying something like, "We couldn't find any revisions o.O"?

Again, some sort of error handling would be good to think about when we see partial functions like these :)

Reply:
Same question here about whether we'll address these here.